synapse/lib/certdir.py

```diff
@@ -825,6 +831,44 @@ def signUserCsr(self, xcsr, signas, outp=None):
         name = xcsr.get_subject().CN
         return self.genUserCert(name, csr=pkey, signas=signas, outp=outp)
 
+    def _loadCasIntoSSLContext(self, ctx):
+        path = s_common.genpath(self.certdir, 'cas')
+        for name in os.listdir(path):
+            if name.endswith('.crt'):
+                ctx.load_verify_locations(os.path.join(path, name))
+
+    def getClientSSLContext(self) -> ssl.SSLContext:
+        '''
+        Returns an ssl.SSLContext appropriate for initiating a TLS session
+        '''
+        sslctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+        self._loadCasIntoSSLContext(sslctx)
+
+        return sslctx
+
+    def getServerSSLContext(self, hostname: Optional[str] = None) -> ssl.SSLContext:
```

Comment:
It was official in 3.7. There are always going to be things that parse comments that are unofficial. They support inline or as a separate file (mainly so that you can annotate 3rd party code).